torchio/transforms/transform.py

```diff
@@ -62,7 +74,22 @@ def __call__(self, data: TypeTransformInput) -> TypeTransformInput:
                 a tensor, the affine matrix is an identity and a tensor will be
                 also returned.
         """
+        if not seed:
+            seed = gen_seed()
+
+        # Store the current rng_state to reset it after the execution
+        torch_rng_state = torch.random.get_rng_state()
+        if isinstance(seed, int):
+            torch.manual_seed(seed=seed)
+
+        self.transform_params = {}
+        self._store_params()
+        self.transform_params['seed'] = seed
+
         if torch.rand(1).item() > self.probability:
+            # If not a Compose
```

Comment:
What if it is `OneOf`?

And why is the transform added to the history if it's not applied to the subject?

Reply:
I thought it would still be useful to keep the information that a transform was not executed because of the random probability, but I cannot think of a specific case where it would be, so I will delete this one.

I am currently working on handling the `OneOf` case, I will update this PR as soon as I get it working

Reply:
As I said above, maybe there's no need to work on reproducibility for `Compose` and `OneOf`. Iterating over the history should do.

Reply:
Clearly, yes...

Reply:
Yes, and it should not be store in the history either this will make the history cleaner than we have now